compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1968,16 +1968,18 @@ object messages {
     }
   }
 
-  case class BadSymbolicReference(
-    location: String,
-    name: String,
-    denotationOwner: Symbol,
-    src: String
-  )(implicit ctx: Context) extends Message(BadSymbolicReferenceID) {
+  case class BadSymbolicReference(denot: SymDenotation)(implicit ctx: Context) extends Message(BadSymbolicReferenceID) {
     val kind = "Reference"
+
+    private val file = denot.symbol.associatedFile
+    private val (location, src) =
+      if (file != null) (s" in $file", file.toString)
+      else ("", "the signature")
+    private val name = ctx.fresh.setSetting(ctx.settings.YdebugNames, true).nameString(denot.name)
```

Comment:
We usually don't create fields to only store temporary values. If you need to precompute something for `msg` , you can use a block:
```scala
val msg = {
  val tmp1 = ...
  val tmp2 = ...
  doSomethingWith(tmp1, tmp2)
}
```
If a temporary computation is shared like `name`, we usually make it a `def`. We might want to create a field if we don't want to perform a computation multiple times. In this case I guess it make sense to keep `name` a field since `ctx.fresh...` may be expensive.